Sync map and scene viewpoints

Rachael-E opened the pull request:
@tschie, please could you review this sample for the sync map and scene viewpoints? Thanks! I'll add in the ReadMe and screenshot this afternoon.

Rachael-E committed:
Create SyncMapAndSceneViewpoints.java

Initialising sample for syncing map and scene viewpoints

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Set up splitpane to view both map and scene view simultaneously

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Set up viewPointChangeHandlers for Scene and Map views

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

removed unused imports

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Reorder imports, added comments for logic

Rachael-E committed:
Create README.md

Initialise README for sample

Rachael-E committed:
Added image and image path

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/SyncMapAndSceneViewpoints.java:
```diff
@@ -0,0 +1,138 @@
+      Viewpoint geoViewPoint = geoView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE);
+
+      // create a list of geoviews being navigated
+      ArrayList<GeoView> arrayListofGeoView = new ArrayList<>();
```
This list will be created every time the viewpoint changes even though the list never changes. You could create it in the start method and reference it here.

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/SyncMapAndSceneViewpoints.java:
```diff
@@ -0,0 +1,138 @@
+    // add the map view and scene view to the split plane
+    splitPane.getItems().addAll(mapView, sceneView);
+
+    // add a view point changed listener to the map view
```
viewpoint one word

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/SyncMapAndSceneViewpoints.java:
```diff
@@ -0,0 +1,138 @@
+   */
+  private void synchronizeViewpoints(GeoView geoView) {
+
+    // if the user is actively navigating the GeoView, get its current viewpoint and set the other view to the same viewpoint
```
would add "when the user is navigating" to the method javadoc and remove this comment.

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/SyncMapAndSceneViewpoints.java:
```diff
@@ -0,0 +1,138 @@
+    splitPane.getItems().addAll(mapView, sceneView);
+
+    // add a view point changed listener to the map view
+    mapView.addViewpointChangedListener(viewpointChangedEvent -> {
```
For one line lambdas you can remove the curly braces and semi-colon. Also, instead of passing the GeoView to the synchronize method, you could pass the event and then use `getSource` on the event to get the GeoView. 
```java
mapView.addViewpointChangedListener(viewpointChangedEvent -> synchronizeViewpoints(viewpointChangedEvent));
```

Rachael-E replied:
This is really neat: didn't realise you could access the GeoView type in this way, thanks for the tip!

colinanderson replied:
If you are passing the event as the argument you should be able to do something like
```
mapView.addViewpointChangedListener(this::synchronizeViewpoints);
```

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/README.md:
```diff
@@ -0,0 +1,40 @@
+
+<p>The two <code>GeoViews</code> share a common <code>ViewPoint</code>. When navigating in one view, the other view is immediately updated to display the same <code>ViewPoint</code>, resulting in a continuously updated and synchronized view.</p>
+
+<p><img src="SyncMapandSceneViewpoints.png" /></p>
```
For the image, It would be good to zoom out on the scene view so it shows the Earth's curvature. Then you don't need the labels.

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/README.md:
```diff
@@ -0,0 +1,40 @@
+<h1>Sync map and scene viewpoints</h1>
+
+<p>Synchronize the view points between a <code>MapView</code> and a <code>SceneView</code>.</p>
```
viewpoints

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/README.md:
```diff
@@ -0,0 +1,40 @@
+
+<p>Synchronize the view points between a <code>MapView</code> and a <code>SceneView</code>.</p>
+
+<p>The two <code>GeoViews</code> share a common <code>ViewPoint</code>. When navigating in one view, the other view is immediately updated to display the same <code>ViewPoint</code>, resulting in a continuously updated and synchronized view.</p>
```
I think this is implied by "synchronized"

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/README.md:
```diff
@@ -0,0 +1,40 @@
+
+<h2>How it works</h2>
+
+<p><code>MapView</code> and <code>SceneView</code> inherit from the <code>GeoView</code> parent class. When the <code>GeoView</code>'s view point has changed, a listener can be added and allow the view point of the other <code>GeoView</code> to be set and synchronised.
```
Consistency on synchronized or synchronised

tschie commented on src/main/java/com/esri/samples/scene/sync_map_and_scene_viewpoints/README.md:
```diff
@@ -0,0 +1,40 @@
+<ul>
+<li>GeoView</li>
+
+<li>CurrentViewpoint(Viewpoint.Type)</li>
```
No methods in this list for Java. I would put GeoView, Viewpoint, and ViewpointChangedEvent.

tschie reviewed (commented):
Looks good! Just some minor suggestions.

Rachael-E committed:
Updated image for showing curvature of 3D scene better, and edits to readme.

Rachael-E committed:
Update README.md

Made intro description more concise

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Renamed array list of geoviews and made it accessible to whole class

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Make JavaDoc more readable and remove additional comment

Rachael-E committed:
Update SyncMapAndSceneViewpoints.java

Shortened viewPointChangedListener to a one line lambda for mapView and sceneView, and ensured consistency of GeoView reference throughout comments.

tschie committed:
Update README.md

tschie reviewed (approved):
Good to merge. Colin is right, you could do that listener yet another way. Up to you if you want to make the change.
